scripts/build/build-utils.ps1

```diff
@@ -80,9 +99,16 @@ function Invoke-MSBuild (
     $solutionName = Split-Path $solutionPath -leaf
     Write-Header "Building solution ${solutionName}"
 
-    $msbuild_exe = Get-MsBuildPath $msbuildVersion
-    & $msbuild_exe $solutionPath $remainingArgs
-    Test-ExitCode "ERROR: Build FAILED."
+    if (Test-Debug) {
+        $remainingArgs += "/v:detailed"
+    }
+    else {
+        $remainingArgs += "/v:quiet"
+    }
+
+    $msbuildExe = Get-MsBuildPath $msbuildVersion
+    Exec { & $msbuildExe $solutionPath $remainingArgs `
+    } -errorMessage "ERROR: Build FAILED."
```

Comment:
I think this is a neater pattern (i.e. calculate the extra parameter first, and only have Exec line appearing once). There's less opportunity for mistakes in the future e.g. if someone edits the script but only changes one of the _Exec_  lines. But I don't think it's worth changing the other occurrences.

Reply:
I haven't used this pattern for NuGet because it doesn't work for it (I cannot understand why).